pilot/pkg/config/kube/ior/route.go

```diff
@@ -365,10 +365,8 @@ func (r *route) verifyResourceVersions(cfg config.Config) error {
 		return fmt.Errorf("could not find an internal reference to gateway %s/%s", cfg.Namespace, cfg.Name)
 	}
 
-	for _, route := range syncRoute.routes {
```

Comment:
I have a hard time grokking how this change related to the rest of this pr .. could you perhaps explain? Probably due to me being naive with respect to this code base .. but if you anticipate future readers may benefit, perhaps add it as a code-level comment.

Reply:
See the commit message. It's a separate commit.

Reply:
Got it, sorry about the noise.